src/query/QueryManager.jl

```diff
@@ -2,36 +2,52 @@
 
 function match(transaction::AbstractCoreTransaction, query::String, options = Proto.Options())
     db_result =  stream(transaction, QueryManagerRequestBuilder.match_req(query, options))
-    maps = map(x->ConceptMap(x), [entry.query_manager_res_part.match_res_part.answers for entry in db_result])
-    result = reduce(vcat, maps)
+    if db_result !== nothing
+        maps = map(x->ConceptMap(x), [entry.query_manager_res_part.match_res_part.answers for entry in db_result])
+        result = reduce(vcat, maps)
+    else
+        result = nothing
+    end
     return result
 end
 
 function match_aggregate(transaction::AbstractCoreTransaction, query::String, options = Proto.Options())
     db_result =  execute(transaction, QueryManagerRequestBuilder.match_aggregate_req(query, options))
-    result = [_read_proto_number(entry.query_manager_res.match_aggregate_res.answer) for entry in db_result]
+    result = _read_proto_number(db_result.query_manager_res.match_aggregate_res.answer)
     return result
 end
 
 function match_group(transaction::AbstractCoreTransaction, query::String, options = Proto.Options())
     db_result =  stream(transaction, QueryManagerRequestBuilder.match_group_req(query, options))
-    maps =  [ConceptMapGroup(item.query_manager_res_part.match_group_res_part.answers) for item in db_result]
-    result = reduce(vcat, maps)
+    if db_result !== nothing
+        maps =  [ConceptMapGroup(item.query_manager_res_part.match_group_res_part.answers) for item in db_result]
```

Comment:
Also, could these two operations be done with a single loop, instead of creating a vector, only to immediately reduce it?
Is each `ConceptMapGroup(item...)` a single value, or can they be vectors themselves?

Reply:
Here each item is a vector. ConceptMapGroup has two constructors one for a single item and one for a vector. The result is a vector of ConceptMapGroup with variable length.

Reply:
Done as you mentioned